swim/test_utils.go

```diff
@@ -121,6 +122,47 @@ func bootstrapNodes(t *testing.T, testNodes ...*testNode) {
 		})
 		require.NoError(t, err, "node must bootstrap successfully")
 	}
+	return hostports
+}
+
+func waitForConvergence(t *testing.T, timeout time.Duration, testNodes ...*testNode) {
+	timeoutCh := time.After(timeout)
+
+	nodes := testNodesToNodes(testNodes)
+
+	// To get the cluster to a converged state we will let the nodes gossip until
+	// there are no more changes. After the cluster finished gossipping we double
+	// check that all nodes have the same checksum for the memberlist, this means
+	// that the cluster is converged.
+	for {
+		select {
+		case <-timeoutCh:
+			t.Errorf("timeout during wait for convergence")
+			return
+		default:
+			for _, node := range nodes {
+				node.gossip.ProtocolPeriod()
+			}
+			hasChanges := false
```

Comment:
Go's standard library has gotos to make control flow easier to read especially in cases where you're using booleans to achieve the same thing: https://golang.org/src/math/gamma.go. I can't promise a dinosaur will not come to eat you though.